src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
+        if (size == elements.length) {
+            Object[] newArray = new Object[Math.round(elements.length * 1.5f)];
+            for (int i = 0; i < elements.length; i++) {
+                newArray[i] = elements[i];
+            }
+            elements = newArray;
+        }
```

Comment:
Consider using System.arraycopy() to copy array elements instead of a manual for loop. This will make the code cleaner and may be more efficient.